src/bespokelabs/curator/request_processor/base_online_request_processor.py

```diff
@@ -472,7 +484,7 @@ async def handle_single_request_with_retries(
                 retry_queue.put_nowait(request)
             else:
                 logger.error(
-                    f"Request {request.task_id} failed permanently after exhausting all 5 retry attempts. "
+                    f"Request {request.task_id} failed permanently after exhausting all {self.max_retries} retry attempts. "
```

Comment:
if require_all_responses is True, shouldn't we terminate here?

Reply:
This is a good question, I think it makes more sense to continue processing as many requests successfully as possible and then failing at the end.

E.g. we are running a data generation job on the cluster and we don't want to to proceeded to the next step (thus raise an error), but we do want it to fufill as many requests as possible so when we restart the job (and it resumes from where it left off) there is less to do.

Reply:
Yeah I am ok with not terminating. I suppose we are mostly trying to give some feedback to the user.